ctapipe/tools/dl1_merge.py

```diff
@@ -0,0 +1,359 @@
+    progress_bar = traits.Bool(help="show progress bar during " "processing").tag(
+        config=True
+    )
+    file_pattern = traits.Unicode(
+        default_value="*.h5", help="Give a specific file pattern for the" "input files"
+    ).tag(config=True)
+
+    parser = ArgumentParser()
+    parser.add_argument("input_files", nargs="*", type=Path)
+
+    aliases = {
+        "input-dir": "MergeTool.input_dir",
+        "output": "MergeTool.output_path",
+        "pattern": "MergeTool.file_pattern",
+    }
+
+    flags = {
+        "skip-images": (
+            {"MergeTool": {"skip_images": True}},
+            "Skip DL1/Event/Telescope and Simulation/Event/Telescope images in output",
+        ),
+        "skip-simu-images": (
+            {"MergeTool": {"skip_simu_images": True}},
+            "Skip Simulation/Event/Telescope images in output",
+        ),
+        "skip-parameters": (
+            {"MergeTool": {"skip_parameters": True}},
+            "Skip DL1/Event/Telescope parameters in output",
+        ),
+        "overwrite": (
+            {"MergeTool": {"overwrite": True}},
+            "Overwrite output file if it exists",
+        ),
+        "progress": (
+            {"MergeTool": {"progress_bar": True}},
+            "show a progress bar during event processing",
+        ),
+    }
+
+    def setup(self):
+        # prepare output path:
+        self.output_path = self.output_path.expanduser()
+        if self.output_path.exists():
+            if self.overwrite:
+                self.log.warning(f"Overwriting {self.output_path}")
+                self.output_path.unlink()
+            else:
+                self.log.critical(
+                    f"Output file {self.output_path} exists, "
+                    "use `--overwrite` to overwrite"
+                )
+                sys.exit(1)
+
+        PROV.add_output_file(str(self.output_path))
+
+        if self.skip_parameters is True and self.skip_images is True:
+            self.log.warning("Skip-parameters and skip-images are both" "set to True")
+
+        # Get input Files
+        args = self.parser.parse_args(self.extra_args)
+        self.input_files = args.input_files
+        if self.input_dir is not None:
+            self.input_files.extend(sorted(self.input_dir.glob(self.file_pattern)))
+        if not self.input_files:
+            self.input_files = sorted(Path(".").glob(self.file_pattern))
+
+        # create output file with subarray from first file
+        self.first_subarray = SubarrayDescription.from_hdf(self.input_files[0])
+        self.first_subarray.to_hdf(self.output_path)
+        self.output_file = tables.open_file(self.output_path, mode="a")
+
+    def check_file(self, file):
+        # Check that the file is not broken or any node is missing
+        file_path = file.root._v_file.filename
+        current_subarray = SubarrayDescription.from_hdf(file_path)
+        # Check subarrays
+        if self.first_subarray != current_subarray:
```

Comment:
I think we should error in this case. We should think about the main use case of this file:

Massive processing of many files on the grid. If a job is successfull, we don't look into the log. So any error should be a real error not just a warning.

Reply:
Maybe an option `--ignore-broken` could be added, to allow users that "know what they are doing"™ to ignore things like these.

But the default case should be an error with a non-zero exit code.

This goes for everything here. Fail loud, fail early. Otherwise errors will get unnoticed on the grid / clusters and subtle errors hard to find are introduced.